common/parallel/FileSystemCrawler.go

```diff
@@ -110,7 +110,11 @@ func Walk(appCtx context.Context, root string, parallelism int, parallelStat boo
 			fsEntry := entry.(FileSystemEntry)
 			err = walkFn(fsEntry.fullPath, fsEntry.info, nil)
 		} else {
-			err = walkFn("", nil, err) // cannot supply path here, because crawlResult probably doesn't have one, due to the error
+			if fsEntry, ok := entry.(FileSystemEntry); ok {
+				err = walkFn(fsEntry.fullPath, fsEntry.info, err)
+			} else {
+				err = walkFn("", nil, err) // cannot supply path here, because crawlResult probably doesn't have one, due to the error
+			}
```

Comment:
add a comment above this
// Our scanner goroutines can enqueue FileSystemEntry items with potentially full path and fileInfo for failures encountered during enumeration. Pass those to caller.